libpromises/generic_agent.c

```diff
@@ -1299,6 +1340,11 @@ bool GenericAgentArePromisesValid(const GenericAgentConfig *config)
         strlcat(cmd, "\"", CF_BUFSIZE);
     }
 
+    if (config->ignore_preferred_augments)
+    {
+        strlcat(cmd, " --ignore-preferred-augments", CF_BUFSIZE);
```

Comment:
Should we not check for truncation?

Reply:
What happens if we get a syntactically valid shell command missing an option?

Reply:
Yea, we should, but the only way for that to happen is if you have a really long bundle sequence. Very unlikely IMO.